bindings/python/src/lib.rs

```diff
@@ -288,25 +306,34 @@ impl PySafeSlice {
 
         let mut offset = 0;
         let length = iterator.remaining_byte_len();
-        let array: PyObject = PyByteArray::new_with(py, length, |bytes: &mut [u8]| {
-            for slice in iterator {
-                let len = slice.len();
-                bytes[offset..offset + slice.len()].copy_from_slice(slice);
-                offset += len;
-            }
-            Ok(())
-        })?
-        .into_py(py);
-        create_tensor(py, &self.framework, self.info.dtype, &newshape, array)
+        let array: PyObject = Python::with_gil(|py| {
+            PyByteArray::new_with(py, length, |bytes: &mut [u8]| {
+                for slice in iterator {
+                    let len = slice.len();
+                    bytes[offset..offset + slice.len()].copy_from_slice(slice);
+                    offset += len;
+                }
+                Ok(())
+            })
+            .unwrap()
+            .into_py(py)
+        });
+        create_tensor(
+            &self.framework,
+            self.info.dtype,
+            &newshape,
+            array,
+            &self.device,
+        )
     }
 }
 
 fn create_tensor(
-    py: Python,
```

Comment:
what is the advantage/reason of removing `py` arg & using `Python::with_gil(|py| {` ?

Reply:
It is the core of the thing.
I must say I don't completely understand but it comes from here:

https://github.com/PyO3/pyo3/issues/1056
https://github.com/PyO3/pyo3/issues/1547

Basically you want to hold the gil as little as possible to give a chance to pyo3 to release it's objects (which then makes the memory available for follow up tensor creation)

Reply:
oh I see 👍